content/en/llm_observability/setup/sdk/nodejs.md

```diff
@@ -0,0 +1,713 @@
+
+`span`
+: optional - _Span_
+<br />The span to extract the span context (span and trace IDs) from. If not provided (as when using function wrappers), the SDK exports the current active span.
+
+#### Example
+
+{{< code-block lang="javascript" >}}
+function llmCall () {
+  const completion = ... // user application logic to invoke LLM
+  const spanContext = llmobs.exportSpan()
+  return completion
+}
+llmCall = llmobs.wrap('llm', { name: 'invokeLLM', modelName: 'claude', modelProvider: 'anthropic' }, llmCall)
+{{< /code-block >}}
+
+
+### Submit evaluations
+
+`llmobs.submitEvaluation()` can be used to submit your custom evaluation associated with a given span.
+
+#### Arguments
+
+The `llmobs.submitEvaluation()` method accepts the following arguments:
+
+`span_context`
+: required - _dictionary_
+<br />The span context to associate the evaluation with. This should be the output of `LLMObs.export_span()`.
+
+`evaluationOptions`
+: required - _object_
+<br />An object of the evaluation data.
+
+The `evaluationOptions` object can contain the following:
+
+`label`
+: required - _string_
+<br />The name of the evaluation.
+
+`metricType`
+: required - _string_
+<br />The type of the evaluation. Must be one of "categorical" or "score".
+
+`value`
+: required - _string or numeric type_
+<br />The value of the evaluation. Must be a string (for categorical `metric_type`) or number (for score `metric_type`).
+
+`tags`
+: optional - _dictionary_
+<br />A dictionary of string key-value pairs that users can add as tags regarding the evaluation. For more information about tags, see [Getting Started with Tags][9].
+
+### Example
+
+{{< code-block lang="javascript" >}}
+function llmCall () {
+  const completion = ... // user application logic to invoke LLM
+  const spanContext = llmobs.exportSpan()
+  llmobs.submitEvaluation(spanContext, {
+    label: "harmfulness",
+    metricType: "score",
+    value: 10,
+    tags: { evaluationProvider: "ragas" }
+  })
+  return completion
+}
+llmCall = llmobs.wrap('llm', { name: 'invokeLLM', modelName: 'claude', modelProvider: 'anthropic' }, llmCall)
+{{< /code-block >}}
+
+## Advanced tracing
+
+### Tracing spans using inline methods
+
+The `llmobs` SDK provides a corresponding inline method to automatically trace the operation a given code block entails. These methods have the same argument signature as their function wrapper counterparts, with the addition that `name` defaults to the span kind (`llm`, `workflow`, and so on) if not provided. This method will finish the span under the following conditions:
+
+1. The function returns a Promise, in which case the span will finish when the promise is resolved or rejected.
+2. The function takes a callback as its last parameter, in which case the span will finish when that callback is called.
+3. The function doesn't accept a callback and doesn't return a Promise, in which case the span will finish at the end of the function execution.
```

Comment:
Clarifying question - is this still accurate for `llmobs.trace()`? I thought inline tracing methods automatically finish the span once you exit the block

Reply:
it's the same here, i think for the underlying tracer it's about consistency. so, if the function returns a promise, the span will finish when the promise is resolved or rejected, not when the promise is returned.